clang/include/clang/Basic/DiagnosticLexKinds.td

```diff
@@ -922,6 +922,10 @@ def err_header_import_semi_in_macro : Error<
 def err_header_import_not_header_unit : Error<
   "header file %0 (aka '%1') cannot be imported because "
   "it is not known to be a header unit">;
+def warn_pp_include_angled_in_module_purview : Warning<
+  "'#include <filename>' in the module purview appears to be erroneous; "
```

Comment:
'appears to be erroneous' seems a bit strong (since we know there are valid cases to do this).  Maybe we can think of a way to explain the actual problem without ending up with too many words (or perhaps have a short warning message and attach a note with explanation)?

Reply:
My thoughts are that all the known valid use cases should use the form `#include "filename"` instead of `#include <filename>`. The angle or the quote matters here. I think it may be a general sense that the headers in the `<>` should be a system header and other things should be put in quotes `""` (I know this is not strictly true). How do you think about this?

Reply:
I agree in general that limiting it to '<...>' seems a reasonable filter; either people should be `import`-ing or `#include "..." ` - but let's also get Michael's opinion.

text-wise - maybe something like:
`#include <xxxx> attaches the declarations to the named module 'M', which is not usually intended;` (where we can name M in the message).

Reply:
Done